### DISCUSSION TITLE
RPi 3A+ --> 44780 displays nothing, null response from encoder, not alot going on here?

### DISCUSSION THREAD

**Discussion by GregWinfield** [probonopd/MiniDexed#512] [Q&A] [answered]
Hey

Built my little Dexed up on a 32GB FAT32 formatted SD in my RPi 3A+
Bot getting a whole heap of anything happening. 
It boots just fine, seeing the Korg Nanokeys.
Screen displays overly contrasted squares, using an I2C backpack on the 44780 LCD. 
Rotary encoder doesn't prompt anything to appear on the display either. 
Everything 'initializes' OK (see screen shot)

Question please ;
- should logging show key presses on the controller, or clicks on the rotary encoder on the HDMI output? if not is there a way i can enable verbose logging to show this?

Pics attached of my wiring, encoder output and HDMI output. 

Boots fine - look OK
![image](https://github.com/probonopd/MiniDexed/assets/130421822/819d2d33-713d-4fc4-87ea-9e019c2003e3)

Wiring
![image](https://github.com/probonopd/MiniDexed/assets/130421822/e180c0b9-0433-40ae-b365-32dd08d37d2b)

My fat hand and display
![image](https://github.com/probonopd/MiniDexed/assets/130421822/7cf73352-760f-4f46-a76b-451afdec678f)

My minidexed.ini
```
#
# minidexed.ini
#

# Sound device
#SoundDevice=i2s
SoundDevice=pwm
#SoundDevice=hdmi
SampleRate=48000
#ChunkSize=256
DACI2CAddress=0
ChannelsSwapped=0

# MIDI
MIDIBaudRate=31250
#MIDIThru=umidi1,ttyS1
MIDIRXProgramChange=1
IgnoreAllNotesOff=0
MIDIAutoVoiceDumpOnPC=1
HeaderlessSysExVoices=0
ExpandPCAcrossBanks=1

# HD44780 LCD
LCDEnabled=1
LCDI2CAddress=0x20

# SSD1306 LCD
# For a 128x32 display, set LCDColumns=20; LCDRows=2
# For a 128x64 display, set LCDColumns=20; LCDRows=4
SSD1306LCDI2CAddress=0x0
SSD1306LCDWidth=128
SSD1306LCDHeight=32
SSD1306LCDRotate=0
SSD1306LCDMirror=0

# Default is 16x2 display (e.g. HD44780)
LCDColumns=16
LCDRows=2

# GPIO Button Navigation
#  Any buttons set to 0 will be ignored
ButtonPinPrev=0
ButtonActionPrev=
ButtonPinNext=0
ButtonActionNext=
ButtonPinBack=11
ButtonActionBack=longpress
ButtonPinSelect=11
ButtonActionSelect=click
ButtonPinHome=11
ButtonActionHome=doubleclick
ButtonPinShortcut=11
# (Shortcut doesn't have an action)

# Timeouts in milliseconds for double click and long press
DoubleClickTimeout=400
LongPressTimeout=400

# MIDI Button Navigation
#  Specify MIDI CC to act as a button
#  NB: Off < 64 < ON
# CC channel: 0=OFF; 1-16 MIDI Ch; >16 Omni
# If MIDIButtonNotes>0 then treat MIDIButton
# numbers as MIDI Note numbers note CC numbers.
MIDIButtonCh=0
MIDIButtonNotes=0
MIDIButtonNotes=1
MIDIButtonPrev=00
MIDIButtonNext=02
MIDIButtonBack=03
MIDIButtonSelect=04
MIDIButtonHome=06

# KY-040 Rotary Encoder
EncoderEnabled=1
EncoderPinClock=10
EncoderPinData=9

# Debug
MIDIDumpEnabled=0
ProfileEnabled=0

# Performance
PerformanceSelectToLoad=1
```

Thoughts/comments very welcome please. Thanks!

**Comment by GregWinfield** [comment]
oh - it does actually work, in that with the korg nanokeys hooked up, it plays notes (albeit via PWM, yes, yuck i know) - but it is working, just the encoder and display don't :)

**Comment by GregWinfield** [comment]
oh, the i2c backpack too is on 0x20 - the jumper config on the back of it - no bridges so all tied low. so life for it begins at 0x20.

**Reply by gerardx40** [reply]
No bridges could also imply 0x27, also see https://lastminuteengineers.com/i2c-lcd-arduino-tutorial/, I think it will vary with the ICs being used, mine is working with 0x27 in a similar setup

**Reply by gerardx40** [reply, marked as answer]
![back](https://github.com/probonopd/MiniDexed/assets/5717319/3355d7d4-10de-4ab4-af36-a161d4e9887c)
![front](https://github.com/probonopd/MiniDexed/assets/5717319/02745bbe-6507-4823-ac1c-5f342e4c4e78)

**Reply by GregWinfield** [reply]
thanks gerard. i'll give that a go before i go down the native raspbian path to see if this thing is in fact talking.

**Reply by GregWinfield** [reply]
thanks gerard. that was it. even though the doco says 0x20. i'll message the supplier too.  LCD is working fine now. thank you so much.

**Comment by GregWinfield** [comment]
plugged the whole shebang into my rpi400 - same result with the screen. just blocks of contrast

**Comment by miotislucifugis** [comment]
try reducing the contrast on the LCD.  do you have a potentiometer on pin#3 of the LCD?

**Reply by GregWinfield** [reply]
hey, yes i've tried that also - the i2c backpack has a pot which dials it back, however still no display. i'm going to check that the i2c and 44780 are in fact working on a vanilla rpi build ; and also confirm 100% that its address is 0x20.

**Comment by Gustavomurta** [comment]
HiGregWinfield,

From the appearance of the image, it seems to me that the Raspberry Pi is not communicating with the LCD Display with I2C interface.
See these procedures to identify the correct I2C address on your module:

https://github.com/probonopd/MiniDexed/discussions/373#discussioncomment-6463794
